program/rust/build.rs

```diff
@@ -1,3 +1,21 @@
+mod build_utils;
+use bindgen;
+use std::collections::HashMap;
+use std::vec::Vec;
+
+
 fn main() {
     println!("cargo:rustc-link-search=../c/target");
+    
+    let borsh_derives: Vec<String> = vec!["BorshSerialize".to_string(), "BorshDeserialize".to_string()];
+
+    let parser = build_utils::DeriveAdderParserCallback{types_to_traits: HashMap::from([
+        ("cmd_hdr", borsh_derives)
+        //map more struct names to traits here, be sure that the definitions of your traites
+        //are included in c_oracle_header.rs
+    ])};
+
+    //generate and write bindings
+    let bindings = bindgen::Builder::default().header("./src/bindings.h").parse_callbacks(Box::new(parser)).generate().expect("Unable to generate bindings");
```

Comment:
nitpic: would be good to run `rustfmt` on all Rust code in this repo as part of the CI/CD to get consistent formatting.

Reply:
oh sorry I have not been doing that for the code I have been pushing, I should probably make a different PR with just cargo fmt after merging this/ other outstanding rust PRs. Otherwise git diff might be confusing

Reply:
I added a all to fmt in  to the Builder in build.rs. Did not seem to change how bindings.rs  lookd much, but  I do not think it's a big problem since you should never have to read it (because it just redefines stuff in oracle.h).

I will start calling cargo fmt on every commit once we make that fmt PR